src/Filter/StringFilter.php

```diff
@@ -129,7 +165,7 @@ public function filter(BaseProxyQueryInterface $query, $alias, $field, $data)
             $parameterName,
             sprintf(
                 $format,
-                $this->getOption('case_sensitive') ? $data['value'] : mb_strtolower($data['value'])
+                $caseSensitive || '' === $data['value'] ? $data['value'] : mb_strtolower($data['value'])
```

Comment:
I haven't checked the failure, but could be because of changing `case_sensitive` to `null` by default, so here it's executed the `mb_strtolower($data['value'])` part and in `PostgreSQL` would not perform the case sensitive query.

Reply:
could change to `false !== $caseSensitive`